### PR TITLE
KNIG-45-Make-fixes-in-ServiceImpl-classes-methods-logic-and-namings

### PR DESCRIPTION
**Pull request by moriarturo**
1. fixed services impls methods logic | 2. fixed namings in services | 3. added findByBlockId method in repository/gateway and its usage in service | 4. moved mapping entity <-> VO to gateway level | 5. added mapping of optionals | 6. services now work only with VO, not directly with entities

### CODE REVIEW

File: src/main/java/com/knighttodo/knighttodo/service/impl/TodoBlockServiceImpl.java [hunk 1]
```diff
@@ -19,54 +16,40 @@
...
 
         if (result.isPresent()) {
-            todoBlock = todoBlockMapper.toTodoBlockVO(result.get());
-        } else {
-            throw new RuntimeException("Did not find TodoBlock id - " + todoBlockId);
+            return result.get();
         }
-
-        return todoBlock;
+        throw new RuntimeException("Did not find TodoBlock id - " + id);
```
Comment (FedosovMax) [inline src/main/java/com/knighttodo/knighttodo/service/impl/TodoBlockServiceImpl.java]: perhaps we'll move it to constants later of maybe do something with localization.

File: src/main/java/com/knighttodo/knighttodo/service/impl/TodoBlockServiceImpl.java [hunk 2]
```diff
@@ -19,54 +16,40 @@
...
             .orElseThrow(TodoNotFoundException::new);
 
-        todoBlock.setId(changedTodoBlockVO.getId());
-        todoBlock.setBlockName(changedTodoBlockVO.getBlockName());
-        todoBlock.setTodos(changedTodoBlockVO.getTodos());
-
-        TodoBlock updatedTodoBlock = todoBlockGateway.save(todoBlock);
-
-        return todoBlockMapper.toTodoBlockVO(updatedTodoBlock);
+        todoBlockVO.setId(changedTodoBlockVO.getId());
```
Comment (FedosovMax) [inline src/main/java/com/knighttodo/knighttodo/service/impl/TodoBlockServiceImpl.java:45]: why don't you use a builder here?

File: src/main/java/com/knighttodo/knighttodo/service/impl/TodoBlockServiceImpl.java (hunk 1 above)
Comment (moriarturo) [reply src/main/java/com/knighttodo/knighttodo/service/impl/TodoBlockServiceImpl.java]: Ok)

File: src/main/java/com/knighttodo/knighttodo/service/impl/TodoBlockServiceImpl.java (hunk 2 above)
Comment (moriarturo) [reply src/main/java/com/knighttodo/knighttodo/service/impl/TodoBlockServiceImpl.java:45]: Cause I think it is not necessary here. To add it we should also add more extra rows. One more reason it that we are not creating a new object here

**Review by FedosovMax — APPROVED** [review]

**Pull request merged**